src/main/scala/pureconfig/WordDelimiter.scala

```diff
@@ -0,0 +1,42 @@
+package pureconfig
+
+trait WordDelimiter {
+  def toTokens(s: String): Seq[String]
```

Comment:
> The use case I have in mind is when users have config schemas they do not control and they want to do fully custom mappings of specific keys.

That's currently possible via a custom `ConfigConvert` for those types, but I'm not against adding that flexibility on the config<->field mappings. @melrief: what do you think?